plasmapy/physics/parameters.py

```diff
@@ -0,0 +1,1486 @@
+        are assumed to be protons.
+
+    Returns
+    -------
+    V_A : Quantity with units of velocity
+        The Alfven velocity of the plasma in units of meters per second.
+
+    Raises
+    ------
+    TypeError
+        The magnetic field and density arguments are not Quantities.
+
+    UnitConversionError
+        If the magnetic field or density is not in appropriate units.
+
+    UserWarning
+        If the Alfven velocity exceeds 10% of the speed of light, or
+        if units are not provided and SI units are assumed.
+
+    ValueError
+        If the density is negative, or the ion mass or charge state
+        cannot be found.
+
+    Notes
+    -----
+    The Alfven velocity :math:`V_A` is the typical propagation speed
+    of magnetic disturbances in a plasma, and is given by:
+
+    .. math::
+    V_A = \frac{B}{\sqrt{\mu_0\rho}}
+
+    where the mass density is :math:`\rho = n_i m_i + n_e m_e`.
+
+    This expression does not account for relativistic effects, and
+    loses validity when the resulting speed is a significant fraction
+    of the speed of light.
+
+    This function switches B and density when B has units of number
+    density or mass density and density has units of magnetic field
+    strength.
+
+    Examples
+    --------
+    >>> from astropy import units as u
+    >>> from plasmapy.constants import m_p, m_e
+    >>> B = 0.014*u.T
+    >>> n = 5e19*u.m**-3
+    >>> rho = n*(m_p+m_e)
+    >>> ion = 'p'
+    >>> Alfven_speed(B, n, ion)
+    <Quantity 43173.871857213584 m / s>
+    >>> Alfven_speed(B, rho, ion)
+    <Quantity 43173.871857213584 m / s>
+    >>> Alfven_speed(rho, B, ion)
+    <Quantity 43173.871857213584 m / s>
+    >>> Alfven_speed(B, rho).to(u.cm/u.us)
+    <Quantity 4.317387185721358 cm / us>
+
+    """
+
+    if isinstance(B, units.Quantity) and isinstance(density, units.Quantity):
+        if B.si.unit in [units.m**-3, units.kg/units.m**3] and \
+                density.si.unit in [units.T]:
+            B, density = density, B
+
+    _check_quantity(B, 'B', 'Alfven_speed', units.T)
+    _check_quantity(density, 'density', 'Alfven_speed',
+                    [units.m**-3, units.kg/units.m**3], can_be_negative=False)
+
+    B = B.to(units.T)
+    density = density.si
+
+    if density.unit == units.m**-3:
+        try:
+            m_i = ion_mass(ion)
+            Z = charge_state(ion)
+            if Z is None:
+                Z = 1
+        except Exception:
+            raise ValueError("Invalid ion in Alfven_speed.")
+        rho = density*m_i + Z*density*m_e
+
+    elif density.unit == units.kg/units.m**3:
+        rho = density
+
+    try:
+        V_A = (np.abs(B)/np.sqrt(mu0*rho)).to(units.m/units.s)
+    except Exception:
+        raise ValueError("Unable to find Alfven speed")
+
+    _check_relativistic(V_A, 'Alfven_speed')
+
+    return V_A
+
+
+def ion_sound_speed(*ignore, ion='p', T_e=0*units.K, T_i=0*units.K,
```

Comment:
My motivation here is to make sure that everything gets called with a keyword, since there's both `T_e` and `T_i`, which means that a call like `ion_sound_speed(5*units.K, 10*units.K)` is ambiguous and confusing.  If there are any positional arguments, then this raises an exception.  Hm...do you think the variable name `ignore` could be changed to something where the motivation is more apparent?